WordPress/Classes/ViewRelated/Reader/Select Interests/ReaderInterestsCollectionViewFlowLayout.swift

```diff
@@ -0,0 +1,198 @@
+            var frame: CGRect = CGRect(origin: .zero, size: itemSize)
+
+            if item == 0 {
+                let minX: CGFloat = isRightToLeft ? maxContentWidth - frame.width : 0
+                frame.origin = CGPoint(x: minX, y: contentInsets.top)
+            } else {
+                frame.origin.x = isRightToLeft ? previousFrame.minX - itemSpacing - frame.width : previousFrame.maxX + itemSpacing
+
+                // If the new X position will go off screen move it to the next row
+                let needsNewRow = isRightToLeft ? frame.origin.x < 0 : frame.maxX > maxContentWidth
+                if needsNewRow {
+                    frame.origin.x = isRightToLeft ? (maxContentWidth - frame.width) : 0
+                    currentRow += 1
+                }
+                frame.origin.y = currentRow * (cellHeight + itemSpacing) + contentInsets.top
+            }
+
+            let attribute = UICollectionViewLayoutAttributes(forCellWith: indexPath)
+            attribute.frame = frame
+
+            layoutAttributes.append(attribute)
+
+            previousFrame = frame
+        }
+
+        // Update content size
+        contentSize.width = maxContentWidth
+        contentSize.height = currentRow * (cellHeight + itemSpacing) + contentInsets.top + contentInsets.bottom
+    }
+
+    override func invalidateLayout() {
+        contentSize = .zero
+        layoutAttributes = []
+
+        super.invalidateLayout()
+    }
+
+    /// Get the size for the given index path from the delegate, or the default item size
+    /// - Parameter indexPath: index path for the item
+    /// - Returns: The width for the cell either from the delegate or the itemSize property
+    private func sizeForItem(at indexPath: IndexPath) -> CGSize {
+        guard
+            let collectionView = collectionView,
+            let delegate = collectionView.delegate as? UICollectionViewDelegateFlowLayout,
+            let size = delegate.collectionView?(collectionView, layout: self, sizeForItemAt: indexPath)
+        else {
+            return CGSize(width: itemSize.width, height: cellHeight)
+        }
+
+        return CGSize(width: size.width, height: cellHeight)
+    }
+
+    override open func layoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        if collectionView?.traitCollection.horizontalSizeClass == .regular {
+            return centeredLayoutAttributesForElements(in: rect)
+        }
+
+        var layoutAttributes = [UICollectionViewLayoutAttributes]()
+
+        for attribute in self.layoutAttributes {
+            if attribute.frame.intersects(rect) {
+                layoutAttributes.append(attribute)
+            }
+        }
+
+        return layoutAttributes
+    }
+
+    private func centeredLayoutAttributesForElements(in rect: CGRect) -> [UICollectionViewLayoutAttributes]? {
+        var rows = [Row]()
+        var rowY: CGFloat = CGFloat.greatestFiniteMagnitude
```

Comment:
Since you declare the type This could be just `var rowY: CGFloat =.greatestFiniteMagnitude`